manager/controlapi/service.go

```diff
@@ -279,6 +279,25 @@ func (s *Server) checkPortConflicts(spec *api.ServiceSpec, serviceID string) err
 	return nil
 }
 
+// checkSecretConflicts finds if the passed in spec has secrets with conflicting targets.
+func (s *Server) checkSecretConflicts(spec *api.ServiceSpec) error {
+	container := spec.Task.GetContainer()
+	if container == nil {
+		return nil
+	}
+
+	existingTargets := make(map[string]string)
+	for _, secretRef := range container.Secrets {
+		if prevSecretName, ok := existingTargets[secretRef.Target]; ok {
+			return grpc.Errorf(codes.InvalidArgument, "secrets '%s' and '%s' have conflicting targets: '%s'", prevSecretName, secretRef.SecretName, secretRef.Target)
```

Comment:
Extremely pedantic nit: should `target` be singular since it's the same target in this case?

Also, should we say the secret references conflict instead of saying the secrets conflict?